spec/javascript/utils/utils.test.js

```diff
@@ -21,3 +34,30 @@ describe('convertCurrency', () => {
     expect(updatedFormValues.lease.another_nested_level.additional_rent).toEqual(425.99)
   })
 })
+
+describe('filterChanged', () => {
```

Comment:
I just thought of a few cases and I'm not sure what the behavior will be, can you add them?

1) a test to check that IDs will be updated properly when entire objects are swapped
```
prevApp = {
  id: 'appid',
  primary_applicant: {
    id: 'primary1',
    name: 'joe',
  }
}

newApp = {
  id: 'appid',
  primary_applicant: {
    id: 'primary2',
    name: 'james',
  }
}

expectedFilteredApp = { 
  id: 'appid',
  primary_applicant: {
    id: 'primary2',
    name: 'james',
  }
}
```

2) a test with arrays
```
prevApp = {
  id: 'appid',
  applicant_contacts: [
   {
      id: 'contact1',
      name: 'joe',
    },
   {
      id: 'contact2',
      name: 'james',
    },
  ],
  applicant_contacts_unchanged: [
   {
      id: 'contact1',
      name: 'joe',
    },
   {
      id: 'contact2',
      name: 'james',
    },
  ],
  other_changed_list: [ 1, 2, 3 ],
  other_unchanged_list: [ 1, 2, 3 ],
}

newApp = {
  id: 'appid',
  applicant_contacts: [
   {
      id: 'contact3',
      name: 'chris',
    },
   {
      id: 'contact4',
      name: 'amy',
    },
  ],
  applicant_contacts_unchanged: [
   {
      id: 'contact1',
      name: 'joe',
    },
   {
      id: 'contact2',
      name: 'james',
    },
  ],
  other_changed_list: [ 1, 2, 3, 4 ],
  other_unchanged_list: [ 1, 2, 3 ],
}

expectedFilteredApp = {
  id: 'appid',
  applicant_contacts: [
   {
      id: 'contact3',
      name: 'chris',
    },
   {
      id: 'contact4',
      name: 'amy',
    },
  ],
  other_changed_list: [ 1, 2, 3, 4 ],
}
```

3: empty object
```
prevApp = { }
newApp = { }
expectedFilteredApp: { }
```

4: object without an id and no changes
```
prevApp = { a: '1' }
newApp = { a: '1' }
expectedFilteredApp: { }
```

5: object without an id and changes
```
prevApp = { a: '1' }
newApp = { a: '2' }
expectedFilteredApp: { a: '2' }
```

6: nested object without changes (when parent does have changes)
```
prevApp = { id: 'abc', a: '1', b: { id: 'abcd', c: '2' } }
newApp = { id: 'abc', a: '2', b: { id: 'abcd', c: '2' } }
expectedFilteredApp: { id: 'abc', a: '2' }
```

7: nested object with changes (when parent does not have other changes)
```
prevApp = { id: 'abc', a: '1', b: { id: 'abcd', c: '2' } }
newApp = { id: 'abc', a: '1', b: { id: 'abcd', c: '3' } }
expectedFilteredApp: { id: 'abc', b: { id: 'abcd', c: '3' } }
```

Reply:
👍